### PR TITLE
Add cmds

### PR DESCRIPTION
**Pull request by pmaslana**
This is just an iterative step to create code needed for a keybase bot to send alerts for stale or pending PRs. Here are the conditions requested from Gene: 

1. If a PR has a new push from an outside contributor and no one has allowed the workflow run to start in say 2 hours - it would post an alert in a Keybase channel.

2. If a community PR has no update from us in 7 days - also post an alert? Rules around this subject to some more thought but.

### CODE REVIEW

File: cmd/notifyPendingCI.go
Comment (cmmarslender) [reply cmd/notifyPendingCI.go:34]: Probably true.. I introduced both of these vars in my PR, thinking it was more clear that adding `--loop` arg to the command meant to loop (and I assumed `loop-time` would rarely be overridden)

Just passing `--loop-time` sort of COULD imply that we always loop, and just vary the time (at least, that was my thinking)

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:39]: This would get PRs as they're created but I think ignores community PRs that have had new commits published to them that are awaiting maintainer approval for CI.

File: cmd/notifyPendingCI.go
Comment (Starttoaster) [reply cmd/notifyPendingCI.go:34]: I think if you don't pass `--loop-time` you'd still get a `loopDuration` variable with an empty/zero duration

Comment (cmmarslender) [reply cmd/notifyPendingCI.go:34]: Loop time defaults to 60 minutes here: https://github.com/Chia-Network/github-bot/blob/main/cmd/root.go#L39

(also being used already in the label community PRs command)

Comment (Starttoaster) [reply cmd/notifyPendingCI.go:34]: Ah, I guess then it makes sense to have two controls

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [inline internal/github/checkStalePRs.go:30]: Sort of a nitpick but duplicate logic could just be its own function

Comment (Starttoaster) [reply internal/github/checkStalePRs.go:30]: Same logic seen in [internal/github/checkPendingCI.go](https://github.com/Chia-Network/github-bot/pull/9/files#diff-9c30dc50701d173ae90de49d485d9031faf7087a7b7359705cc4dee52bea329a)

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [inline internal/github/checkStalePRs.go:59]: I'll have to look, but I'm curious if there are other types of events that should count here.

Comment (Starttoaster) [reply internal/github/checkStalePRs.go:59]: Maybe https://docs.github.com/en/rest/using-the-rest-api/issue-event-types?apiVersion=2022-11-28#reviewed

Comment (Starttoaster) [reply internal/github/checkStalePRs.go:59]: Though, I don't know if `reviewed` maybe also implies `commented`? But I don't know if it necessarily does or doesn't

**Review by Starttoaster — COMMENTED** [review]
In addition to my other comments, maybe also do a `go get -u ./... && go mod tidy`, the dependency review job is upset https://github.com/Chia-Network/github-bot/actions/runs/8944275086/job/24570869218#step:3:7

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [reply internal/github/checkStalePRs.go:59]: Additionally, with regards to reviews, if the last event IS a review, maybe this stale PR should also just be ignored? That might be up for debate tbh. But my logic there is basically, if we're waiting on the PR author to submit requested changes, the ball isn't waiting in our court.

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:20]: `cfg.LabelCheckRepos` -- should this have its own repo list actually? It's a different check than the label checks, which has me leaning towards "probably?"

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [inline internal/github/checkStalePRs.go:24]: `cfg.LabelCheckRepos` should this be its own list in the config file since it's a different check than the label one?

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go:59]: Maybe commented or reviewed by a teammember within the last seven days then?

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go:30]: I thought of this while adding the code here and thought that this might come up. I will go ahead and make this a helper function.

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:20]: I added an additional config section for listing repos that will checked for stale/pending PRs

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go]: I added a check to see if there was a dismissal for a prior approval. Also, if a commit is made it will change the cutofftime.

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [reply internal/github/checkStalePRs.go:47]: Yeah that makes sense to me.

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go]: I did integrate this the helper. But, I also needed that logic in checkPendingCI.go because I need the owner repo variables.

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go]: Added checks for if a pr has been reviewed or not

File: k8s/notify_stale_pending_prs.yml.j2
Comment (cmmarslender) [inline k8s/notify_stale_pending_prs.yml.j2:8]: This action should be notify-pending or something right? Label-prs is my command

File: internal/github/checkPendingCI.go
Comment (cmmarslender) [inline internal/github/checkPendingCI.go:109]: I think checking for dismissed probably makes sense.. but we should probably check if dismissed and doesn't have an approval from someone else

File: internal/github/checkStalePRs.go
Comment (cmmarslender) [inline internal/github/checkStalePRs.go:43]: I believe this is limiting this whole function to 30 seconds? What if there are a lot of pages to fetch and it takes longer than that?

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: internal/github/pullRequests.go
Comment (cmmarslender) [inline internal/github/pullRequests.go:44]: If we're going to be checking for the minimum PR number in the API call loop, you can actually `break` here to svae on API requests and move on to the next repo, because the API request options say to sort the PRs by created, descending, so the PR numbers should always be descending.

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:109]: What about checking to see the status of the most recent review? If that is "DISMISSED" then return true.

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go:43]: Changed to 300s

File: k8s/notify_stale_pending_prs.yml.j2
Comment (pmaslana) [reply k8s/notify_stale_pending_prs.yml.j2]: Added the appropriate commands

File: internal/github/pullRequests.go
Comment (pmaslana) [reply internal/github/pullRequests.go]: Changed the continue to a break

File: cmd/notifyStale.go
Comment (Starttoaster) [inline cmd/notifyStale.go:31]: Do we want the bot to exit here? This seems like it's meant to be an infinite loop. If it fails this loop iteration once, is it guaranteed to fail again?

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:25]: If someone gives an invalid repo name to the config, do we want it to completely stop here, like it currently is? Or can we continue on to the next repo in the config?

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:45]: This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:51]: This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:57]: This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

File: internal/github/checkPendingCI.go
Comment (Starttoaster) [inline internal/github/checkPendingCI.go:61]: This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

File: internal/github/checkStalePRs.go
Comment (Starttoaster) [inline internal/github/checkStalePRs.go:30]: This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

File: internal/label/pullrequests.go
Comment (Starttoaster) [inline internal/label/pullrequests.go:62]: This doesn't really need a comment imo. Seems pretty clear what's happening here, the comment is fluff.

File: cmd/notifyStale.go
Comment (pmaslana) [reply cmd/notifyStale.go:31]: Fixed

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:25]: I added a continue and changed to printing out the error

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:39]: added owner/repo

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:45]: added owner/repo

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:51]: added owner/repo

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:57]: added owner/repo

File: internal/github/checkPendingCI.go
Comment (pmaslana) [reply internal/github/checkPendingCI.go:61]: added owner/repo

File: internal/github/checkStalePRs.go
Comment (pmaslana) [reply internal/github/checkStalePRs.go:30]: Added the reponame

File: internal/label/pullrequests.go
Comment (pmaslana) [reply internal/label/pullrequests.go:62]: Removed

File: internal/github/checkStalePRs.go
Comment (cmmarslender) [reply internal/github/checkStalePRs.go]: > Changed to 300s

The point I was getting at is that a timeout for a loop of unknown length is probably not the route we want to take. If your set on keeping a context here, then it should likely just be a context for each request, if the requests don't support timeouts more directly

File: cmd/notifyPendingCI.go
Comment (Starttoaster) [inline cmd/notifyPendingCI.go:33]:
```suggestion
				time.Sleep(loopDuration)
				continue
```

Comment (Starttoaster) [reply cmd/notifyPendingCI.go:33]: This continue circumvents the sleep time below. You probably don't want to retry all your github API requests right away if this failed here.

File: cmd/notifyStale.go
Comment (Starttoaster) [inline cmd/notifyStale.go:33]:
```suggestion
				time.Sleep(loopDuration)
				continue
```
This continue circumvents the sleep time below. You probably don't want to retry all your github API requests right away if this failed here.

File: cmd/notifyPendingCI.go
Comment (pmaslana) [reply cmd/notifyPendingCI.go:34]: added

**Review by Starttoaster — APPROVED** [review]

**Pull request merged**
